src/GafferUI/StandardNodeGadget.cpp

```diff
@@ -1133,9 +1137,24 @@ void StandardNodeGadget::enter( Gadget *gadget )
 {
 	if( m_labelsVisibleOnHover )
 	{
+		IECore::ConstBoolDataPtr inputVisibleData = Gaffer::Metadata::value<IECore::BoolData>( node(), g_inputNoduleLabelsVisibleKey );
+		IECore::ConstBoolDataPtr outputVisibleData = Gaffer::Metadata::value<IECore::BoolData>( node(), g_outputNoduleLabelsVisibleKey );
+
 		for( StandardNodule::RecursiveIterator it( gadget  ); !it.done(); ++it )
 		{
-			(*it)->setLabelVisible( true );
+			if(
+				(
+					(*it)->plug()->direction() == Plug::Direction::In &&
+					( !inputVisibleData || inputVisibleData->readable() )
+				) ||
+				(
+					(*it)->plug()->direction() == Plug::Direction::Out &&
+					( !outputVisibleData || outputVisibleData->readable() )
+				)
+			)
+			{
+				(*it)->setLabelVisible( true );
+			}
```

Comment:
I'm not sure about this on two counts :

- I think it's more useful if `m_labelsVisibleOnHover` trumps `nodeGadget:inputNoduleLabelsVisible`. Without that you're almost completely in the dark as to what you're connecting to when `inputNoduleLabelsVisible` is False.
- `inputNoduleLabelsVisible == None` is acting like `inputNoduleLabelsVisible == True` here, but that's the opposite of how it acts in `updatePersistentNoduleLabels()`.

I think we get more useful and consistent behaviour if we just revert to the original code here.

Reply:
Reverted and squashed down into https://github.com/GafferHQ/gaffer/pull/6135/commits/7ec405e6f2e7789e74760eaf2695c9d70f2b811a (hopefully that's clearer than adding a new fixup to revert).